galley/pkg/config/analysis/analyzers/service/portname.go

```diff
@@ -72,6 +72,11 @@ func (s *PortNameAnalyzer) analyzeService(r *resource.Instance, c analysis.Conte
 			m := msg.NewPortNameIsNotUnderNamingConvention(
 				r, port.Name, int(port.Port), port.TargetPort.String())
 
+			if svc.Type == "ExternalName" {
+				m = msg.NewExternalNameServiceTypeInvalidPortName(
+					r, port.Name, int(port.Port), port.TargetPort.String())
+			}
```

Comment:
I'm thinking only to generate the error for port 443 since it is a common protocol for HTTPS.  Declaring `protocol: tcp` could be valid if tcp is the actual protocol, for example on port 9090.